### PR TITLE
refactor(lazy-pages): use `LimitedStr` for global names

### LINKED ISSUE #2098: use `TrimmedString` for global names in `GlobalsCtx`
### File Location(s)

[core-backend/common/src/lazy_pages.rs](https://github.com/gear-tech/gear/pull/1988/files/55e7748b7428483d5594328b4ad0295e2b312d5f#diff-37fb51e19dec925a94d0609cc0723eabd2b3d8d7759aaad92e883ec2b786e63a)

### Proposal

__IMPORTANT__: need to implement `TryFrom` for `TrimmedString`, using `From` is RESTRICTED!!!!

### PR DESCRIPTION
**Pull request by tltsutltsu**
Resolves #2098.

### CODE REVIEW

**Comment by SabaunT** [comment]
@breathx what's the rationale of using a heap-allocated value here?

File: lazy-pages/src/globals.rs [hunk 1]
```diff
@@ -94,7 +97,10 @@ fn apply_for_global_internal(
     name: &str,
     mut f: impl FnMut(u64) -> Result<Option<u64>, Error>,
 ) -> Result<u64, Error> {
-    let current_value = globals_access_provider.get_i64(name)? as u64;
+    let name =
+        TrimmedString::try_from(name).map_err(|_| Error::AccessGlobal(GlobalsAccessError))?;
+
+    let current_value = globals_access_provider.get_i64(name.clone())? as u64;
```
[the remark this answers is not in the document]
Comment (tltsutltsu) [reply lazy-pages/src/globals.rs]: Sure. Thanks!

**Review by shamilsan — APPROVED** [review]
@grishasobol Take a look, please. Formally, the issue has been solved, but is it the right thing you meant in your task?

**Comment by breathx** [comment]
> @breathx what's the rationale of using a heap-allocated value here?

Makes sense to avoid heap allocations, but then needs to provide some wrapper maybe? `struct LimitedStr<'a>(&'a str)`, that would restrict huge names?

**Review by breathx — APPROVED** [review]

**Pull request merged**